lib/cosmos/interfaces/protocols/protocol.rb

```diff
@@ -15,9 +15,12 @@
 module Cosmos
   class Protocol
     attr_accessor :interface
+    attr_accessor :allow_empty_data
 
-    def initialize
+    # @param allow_empty_data [true/false] Whether STOP should be returned on empty data
+    def initialize(allow_empty_data = false)
```

Comment:
Its available from OverrideProtocol (and future other protocols).   If you used override protocol by itself, without setting this you could have issues.